docs/content/ir.md

```diff
@@ -0,0 +1,435 @@
+---
+title: Intermediate Representation (IR)
+kind: misc
+weight: 1
+---
+
+# Overview
+
+OPA can compile policy queries into planned evaluation paths suitable for
+further compilation or interpretation. This document explains the structure and
+semantics of the intermediate representation (IR) used to represent these
+planned evaluation paths. Read this document if you want to write a compiler or
+interpreter for Rego.
```

Comment:
Do we need some declaration on how stable what you see here is? In other words, do we want to reserve ourselves some wiggle room in terms of compatibility guarantees? 🤔 

I suppose without any declared "versions" of the IR, we're either in "anything goes" or in "this is never going to change" territory. Just not sure which one it is. 😅

Reply:
I think the most likely changes will be additive, e.g., new statements or sections to support language features and optimizations. In that case, we'll need a way to indicate the plan version to the consumer. I'll update the PR with a version field on the top-level policy struct (a single integer should do the trick?)